### ISSUE TITLE
What is the meaning of operator[](T* key)

### ISSUE THREAD

**Issue by scinart** [nlohmann/json#645]
What's the meaning of the following code?

abbreviated version from json.hpp:

```c++
template<typename T, std::size_t n>
reference operator[](T * (&key)[n])
{
    return operator[](static_cast<const T>(key));
}
template<typename T>
reference operator[](T* key)
{
    return m_value.object->operator[](key);
}
```
and const_reference version of them.

Normally m_value.object is an std::map and key is an std::string. But the code must have been placed for a reason. I just want to know why.

**Comment by nlohmann** [comment]
Good point. I see the code is currently not covered by any test case. I think it was old code that aimed to avoid a conversion from string literals to `string_t`. I wonder why they are not used any more.

**Comment by nlohmann** [comment]
The code was added in commit 7e3245786c63c0923186b2b38c9c874ed2cc3942 to fix issue #171. With current versions of Clang, it seems these functions are not used any more.

**Comment by nlohmann** [comment]
I removed the functions and none of the CI compilers complained. Closing this issue. Thanks @scinart for noting!
